src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_INITIAL_ARRAY_CAPACITY = 10;
+    private Object[] values;
+    private int size;
+    private int capacityOfArray;
+
+    public ArrayList() {
+        capacityOfArray = DEFAULT_INITIAL_ARRAY_CAPACITY;
+        values = new Object[capacityOfArray];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacityOfArray) {
+            growArray();
+        }
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is invalid");
```

Comment:
```suggestion
            throw new ArrayListIndexOutOfBoundsException("Index  "+index+"is invalid for size " + size);
```

lets make exception messages more informative

Reply:
msg expanded